src/renderer/entities/governance/lib/governanceSubscribeService.ts

```diff
@@ -50,118 +53,131 @@ function subscribeVotingFor(
-            votes[referendumId] = {
-              type: 'Split',
-              aye: splitVote.aye.toBn(),
-              nay: splitVote.nay.toBn(),
-            };
+      switch (convictionVoting.type) {
+        case 'Delegating': {
+          result[address][trackId] = {
+            type: 'Delegating',
+            address,
+            track: trackId,
+            balance: convictionVoting.data.balance,
+            conviction: convictionVoting.data.conviction,
+            target: toAddress(convictionVoting.data.target),
+            prior: convictionVoting.data.prior,
+          };
+          break;
+        }
+        case 'Casting': {
+          const votes: Record<ReferendumId, AccountVote> = {};
+          for (const { referendum, vote } of convictionVoting.data.votes) {
+            const referendumId = referendum.toString();
+
+            switch (vote.type) {
+              case 'Standard':
+                votes[referendumId] = {
+                  type: 'Standard',
+                  vote: {
+                    aye: vote.data.vote.isAye,
+                    conviction: vote.data.vote.conviction.type,
+                  },
+                  balance: vote.data.balance,
+                };
+                break;
+              case 'Split':
+                votes[referendumId] = {
+                  type: 'Split',
+                  aye: vote.data.aye,
+                  nay: vote.data.nay,
+                };
+                break;
+              case 'SplitAbstain':
+                votes[referendumId] = {
+                  type: 'SplitAbstain',
+                  aye: vote.data.aye,
+                  nay: vote.data.nay,
+                  abstain: vote.data.abstain,
+                };
+                break;
+            }
           }
 
-          if (vote.isSplitAbstain) {
-            const splitAbstainVote = vote.asSplitAbstain;
-            votes[referendumId] = {
-              type: 'SplitAbstain',
-              aye: splitAbstainVote.aye.toBn(),
-              nay: splitAbstainVote.nay.toBn(),
-              abstain: splitAbstainVote.abstain.toBn(),
-            };
-          }
+          result[address][trackId] = {
+            type: 'Casting',
+            track: trackId,
+            address,
+            votes,
+            prior: convictionVoting.data.prior,
+          };
         }
-
-        result[address][trackId] = {
-          type: 'Casting',
-          track: trackId,
-          address,
-          votes,
-          prior: {
-            unlockAt: convictionVoting.asCasting.prior[0].toNumber(),
-            amount: convictionVoting.asCasting.prior[1].toBn(),
-          },
-        };
       }
     }
 
     callback(result);
   });
+}
 
-  return () => {
-    unsubscribe.then((fn) => fn());
+function subscribeReferendums(api: ApiPromise, callback: (referendums: IteratorResult<Referendum[], void>) => unknown) {
+  let currectAbortController = new AbortController();
```

Comment:
current?

Reply:
On each event `AbortController` breaks current page request, but since we can't reset controller it got recreated every time
So, it's `current`